planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/utils.cpp

```diff
@@ -250,11 +238,47 @@ std::optional<size_t> findIndexOutOfGoalSearchRange(
   return min_dist_out_of_range_index;
 }
 
+using PathIterator =
+  decltype(autoware_internal_planning_msgs::msg::PathWithLaneId::points)::iterator;
+
+void fillLaneIdsFromMap(
+  PathIterator begin, PathIterator end, const std::shared_ptr<RouteHandler> & route_handler)
+{
+  for (auto it = begin; it != end; ++it) {
+    const auto point = it->point;
+    lanelet::ConstLanelet lanelet;
+    if (route_handler->getClosestLaneletWithinRoute(point.pose, &lanelet)) {
```

Comment:
nits:
this is edge case but may get wrong lane from overrlapped route lanes.
better to search from lane sequence after `min_dist_out_of_circle_index` lane id for continuity.